internal/uvm/cpugroups.go

```diff
@@ -0,0 +1,55 @@
+package uvm
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/cpugroup"
+	hcsschema "github.com/Microsoft/hcsshim/internal/schema2"
+)
+
+// ReleaseCPUGroup unsets the cpugroup from the VM
+func (uvm *UtilityVM) ReleaseCPUGroup(ctx context.Context) error {
+	groupID := uvm.cpuGroupID
+	if groupID == "" {
+		// not set, don't try to do anything
+		return nil
+	}
+	if err := uvm.unsetCPUGroup(ctx); err != nil {
+		return fmt.Errorf("failed to remove VM %s from cpugroup %s", uvm.ID(), groupID)
+	}
+	return nil
+}
+
+// SetCPUGroup setups up the cpugroup for the VM with the requested id
+func (uvm *UtilityVM) SetCPUGroup(ctx context.Context, id string) error {
+	if id == "" {
+		return errors.New("must specify an ID to use when configuring a VM's cpugroup")
+	}
+	return uvm.setCPUGroup(ctx, id)
+}
+
+// setCPUGroup sets the VM's cpugroup
+func (uvm *UtilityVM) setCPUGroup(ctx context.Context, id string) error {
+	req := &hcsschema.ModifySettingRequest{
+		ResourcePath: cpuGroupResourcePath,
+		Settings: &hcsschema.CpuGroup{
+			Id: id,
+		},
+	}
+	if err := uvm.modify(ctx, req); err != nil {
+		return err
+	}
+	uvm.cpuGroupID = id
```

Comment:
do you need a lock here?

Reply:
Well setting the CPU group is only done on create*COW and unsetting only happens on Release when the UVM is closing. However I think that closing the UVM could be called multiple times which would call Release multiple times potentially, but I'm not sure if that would really be an issue. What do you think?

Reply:
ok, I see. it's probably unlikely then. 👍